### PR TITLE
add Annotation related descriptions

### PR DESCRIPTION
**Pull request by armintaenzertng**
This is an attempt to fill in the missing descriptions in the Core profile. Where possible, I tried to copy from the SPDX-2.3 specification.
Please note that this does not aim to be the final solution but will hopefully start discussions on possibly controversial points.

### CODE REVIEW

File: model/Core/Vocabularies/AnnotationType.md [hunk 1]
```diff
@@ -4,18 +4,17 @@ SPDX-License-Identifier: Community-Spec-1.0
...
 ## Metadata
 
 - name: AnnotationType
 
 ## Entries
 
-- other: TODOdescription
-- review: TODOdescription
-
+- other: Used to store extra information about an Element during creation.
```
Comment (goneall) [inline model/Core/Vocabularies/AnnotationType.md:19]: Although this generally follows the 2.3 documentation, the other type may also be used after the creation.  Suggest softening this a bit like: 

`Used to store extra information about an Element which is not part of a Review (e.g. extra information provided during the creation of the Element).`

**Review by goneall — COMMENTED** [review]
One minor suggestion

File: model/Core/Vocabularies/AnnotationType.md (hunk 1 above)
Comment (armintaenzertng) [reply model/Core/Vocabularies/AnnotationType.md]: done

**Review by goneall — APPROVED** [review]
LGTM

**Comment by goneall** [comment]
This PR only adds missing text, merging per the process discussed on 28 Mar 2023 tech call

**Pull request merged**